diff --git a/assets/scripts/recipes/events.js b/assets/scripts/recipes/events.js
index 883f1c4..4212834 100644
--- a/assets/scripts/recipes/events.js
+++ b/assets/scripts/recipes/events.js
@@ -36,8 +36,8 @@ const onDeleteRecipe = (event) => {
 
 const onUpdateRecipe = (event) => {
   event.preventDefault()
-  const recipeId = $(event.target).closest('.update-recipe-forms').data('id')
   const form = event.target
+  const recipeId = $(form).data('id')
   const formData = getFormFields(form)
   api.updateRecipes(formData, recipeId)
     .then(ui.updateRecipesSuccess)
diff --git a/assets/scripts/recipes/ui.js b/assets/scripts/recipes/ui.js
index 902fc38..a770172 100644
--- a/assets/scripts/recipes/ui.js
+++ b/assets/scripts/recipes/ui.js
@@ -5,6 +5,7 @@ const showRecipesTemplate = require('../templates/recipe-posting.handlebars')
 const getRecipesSuccess = (data) => {
   const showRecipesHtml = showRecipesTemplate({ recipes: data.recipes })
   $('.content').html(showRecipesHtml)
+  $("body").removeAttr('class')
   $('#message').text('Here are your recipes!')
 }
 
@@ -31,7 +32,7 @@ const deleteRecipeFailure = (id) => {
 const updateRecipesSuccess = (id) => {
   $('#message-2').text('You updated a recipe!')
   $('form').trigger('reset')
-  $('#update-modal').modal('hide')
+  $('.modal').modal('hide')
   $('.modal-backdrop').remove()
 }
 
diff --git a/assets/scripts/templates/recipe-posting.handlebars b/assets/scripts/templates/recipe-posting.handlebars
index 5a4a14c..035f513 100644
--- a/assets/scripts/templates/recipe-posting.handlebars
+++ b/assets/scripts/templates/recipe-posting.handlebars
@@ -18,27 +18,27 @@
     <h3 class="font-weight-bold">Directions:</h3>
     <p class="font-italic">{{recipe.directions}}</p>
     <button type="click" data-id="{{recipe._id}}" class="btn btn-danger delete-recipe">Delete this recipe</button>
-    <button type="button" class="btn btn-secondary" data-toggle="modal" data-target="#update-modal">
+    <button type="button" class="btn btn-secondary" data-toggle="modal" data-target="#update-modal-{{recipe._id}}">
       Update this recipe
     </button>
     <!-- Modal -->
-    <div class="modal fade" id="update-modal" aria-labelledby="recipeModalLabel" aria-hidden="true">
+    <div class="modal fade" id="update-modal-{{recipe._id}}" aria-labelledby="recipeModalLabel-{{recipe._id}}" aria-hidden="true">
       <div class="modal-dialog">
         <div class="modal-content">
           <div class="modal-header">
-            <h5 class="modal-title" id="recipeModalLabel">Updating an existing recipe?</h5>
+            <h5 class="modal-title" id="recipeModalLabel-{{recipe._id}}">Updating an existing recipe?</h5>
             <button type="button" class="close" data-dismiss="modal" aria-label="Close">
               <span aria-hidden="true">&times;</span>
             </button>
           </div>
           <div class="modal-body">
             <form class="update-recipe-forms form" data-id="{{recipe._id}}">
-              <input name="recipe[ingredient]" type="text" placeholder="ingredient">
+              <input name="recipe[ingredient]" type="text" placeholder="{{recipe.ingredient}}">
               <input name="recipe[temperature]" type="number" placeholder="temperature in &deg F">
               <input name="recipe[time]" type="number" placeholder="time in hours">
               <input name="recipe[fuel]" type="text" placeholder="fuel source">
               <input name="recipe[directions]" type="text" placeholder="directions">
-              <button type="submit" data-toggle="modal" data-target="#update-modal" class="btn btn-success">Save changes</button>
+              <button type="submit" data-toggle="modal" data-target="update-modal-{{recipe._id}}" class="btn btn-success">Save changes</button>
             </form>
           </div>
           <div class="modal-footer">
